python/pylibcudf/tests/test_column_from_iterable.py

```diff
@@ -216,3 +216,36 @@ def test_from_nested_list_of_large_strings(dummy_large_string_type):
     assert col.type().id() == plc.TypeId.LIST
     assert col.children()[1].type().id() == plc.TypeId.STRING
     assert col.children()[1].children()[0].type().id() == plc.TypeId.INT64
+
+
+@pytest.mark.parametrize(
+    "expect, dtype",
+    [
+        ([-1, 2, 3], plc.DataType(plc.TypeId.INT8)),
+        ([-1, 2, 3], plc.DataType(plc.TypeId.INT16)),
+        ([-1, 2, 3], plc.DataType(plc.TypeId.INT32)),
+        ([-1, 2, 3], plc.DataType(plc.TypeId.INT64)),
+        ([1, 2, 3], plc.DataType(plc.TypeId.UINT8)),
+        ([1, 2, 3], plc.DataType(plc.TypeId.UINT16)),
+        ([1, 2, 3], plc.DataType(plc.TypeId.UINT32)),
+        ([1, 2, 3], plc.DataType(plc.TypeId.UINT64)),
+        ([1.25, 4.5, 7.75], plc.DataType(plc.TypeId.FLOAT32)),
+        ([1.25, 4.5, 7.75], plc.DataType(plc.TypeId.FLOAT64)),
+        ([True, False, True], plc.DataType(plc.TypeId.BOOL8)),
+        (["foo", "bar", "baz"], plc.DataType(plc.TypeId.STRING)),
```

Comment:
Could you add a test for that includes some chars that are multiple bytes when utf-8 encoded? Just since we're indexing into the buffer, it'd be good to have a test case that exercises that a bit.

And would it be possible to also parametrize this by some null handling? Maybe a new fixture that parametrizes over `[True, False]` for whether to set the first item to null?

I also see some special treatment of length-0 columns. It'd be good to test that too (here, or in another test; whatever is easiest).

Finally, you've probably gotten the bitmask stuff correct, but even if you parametrize this test by whether there are any nulls, we'd still just be dealing with the first byte of the bitmask (I think). Maybe a separate test with some longer data?